dask/array/core.py

```diff
@@ -1187,7 +1192,8 @@ def store(
         layers[targets_name] = targets_layer
         dependencies[targets_name] = set()
 
-    load_stored = return_stored and not compute
+    if load_stored is None:
+        load_stored = return_stored and not compute
```

Comment:
@jakirkham i was very confused by the default for `load_stored`. Why `not compute`? It would be useful to add a comment for the next person who comes around :)

Reply:
Perhaps it helps to read the discussion in PR: https://github.com/dask/dask/pull/3153